QWeb/internal/xhr.py

```diff
@@ -16,12 +16,256 @@
+      XMLHttpRequest.prototype.send = function() {
+        if (this.__xhrTracked) {
+          try { window.__xhrMon.pending++; } catch(e) {}
+          this.addEventListener("loadend", function(){ try { window.__xhrMon.pending--; } catch(e) {} }, { once:true });
+        }
+        return _send.apply(this, arguments);
+      };
+      window.__xhrMon.xhrPatched = true;
+    }
+  } catch(e){}
+
+  try {
+    if (!window.__xhrMon.observerStarted && window.MutationObserver) {
+      const obs = new MutationObserver(function(){ window.__xhrMon.lastMutationTs = Date.now(); });
+      obs.observe(document.documentElement || document.body, {
+        childList:true, subtree:true
+      });
+      window.__xhrMon.observerStarted = true;
+    }
+  } catch(e){}
+
+  return true;
+})();
+"""
+# This was modified above:
+# obs.observe(document.documentElement || document.body, {
+#        childList:true, subtree:true, attributes:true, characterData:true
+#      });
+
+# Lightweight status (no spinner logic here anymore)
+JS_STATUS_LITE = r"""
+return (function (quietMs) {
+  var ready = (document.readyState === "complete");
+
+  var jqActive = null;
+  try {
+    if (typeof jQuery !== "undefined" && jQuery && typeof jQuery.active === "number") jqActive = jQuery.active;
+  } catch(e){}
+
+  var pending = 0, last = Date.now();
+  try {
+    pending = (window.__xhrMon && typeof window.__xhrMon.pending === "number") ? window.__xhrMon.pending : 0;
+    last    = (window.__xhrMon && window.__xhrMon.lastMutationTs) ? window.__xhrMon.lastMutationTs : last;
+  } catch(e){}
+
+  var now = Date.now();
+  var domQuiet = (now - last) >= (quietMs || 400);
+  var networkIdle = (pending === 0) && (jqActive === null || jqActive === 0);
+  var idle = ready && networkIdle && domQuiet;
+
+  return {
+    idle: idle,
+    ready: ready,
+    networkIdle: networkIdle,
+    domQuiet: domQuiet,
+    pending: pending,
+    jqActive: jqActive
+  };
+})(arguments[0]);
+"""
+
+# Separate spinner probe (selectors configurable)
+JS_IS_SPINNER_BUSY = r"""
+return (function (selectors) {
+  function visible(el){
+    if (!el) return false;
+    var cs = getComputedStyle(el);
+    if (cs.visibility==='hidden' || cs.display==='none' || cs.opacity==='0') return false;
+    var r = el.getBoundingClientRect();
+    return r.width>0 && r.height>0;
+  }
+  try {
+    var sels = Array.isArray(selectors) ? selectors : [];
+    for (var i=0;i<sels.length;i++){
+      var nodes = document.querySelectorAll(sels[i]);
+      for (var j=0;j<nodes.length;j++){
+        if (visible(nodes[j])) return true;
+      }
+    }
+    return false;
+  } catch (e) {
+    return false; // if probe fails, treat as not-busy (avoid blocking)
+  }
+})(arguments[0]);
+"""
 
 
 def setup_xhr_monitor() -> bool:
+    try:
+        ok = javascript.execute_javascript(JS_INSTALL_MONITOR)
+        if ok is not True:
```

Comment:
Cast the `ok` to `bool` before checking the identity or remove the cast from `return` statement if `execute_javascript` is guaranteed to return `bool`

```
>>> "true" is not True
<stdin>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
True
```

Reply:
ok